src/ros_filter.cpp

```diff
@@ -950,17 +951,21 @@ void RosFilter::loadParams()
     ss << "odom" << topic_ind++;
     std::string odom_topic_name = ss.str();
     std::string odom_topic;
-    more_params = node_->get_parameter(odom_topic_name, odom_topic);
+    node_->declare_parameter(odom_topic_name);
+
+    rclcpp::Parameter parameter;
```

Comment:
Can you explain what's happening here? I see some gets but no declares and the 2 `get_parameter` calls look identical

Reply:
So my thinking was this: 
`get_parameter` does not cast to bool anymore, and we also need to `declare_parameter` before using it. 

The code 
```
node_->declare_parameter(odom_topic_name);

    rclcpp::Parameter parameter;
    if (rclcpp::PARAMETER_NOT_SET != node_->get_parameter(odom_topic_name, parameter)) {
      more_params = true;
      odom_topic = node_->get_parameter(odom_topic_name, parameter);
    };
```
will first declare an empty `odom_topic_name` which will get rclcpp::PARAMETER_NOT_SET if not provided in the config. 
The conditional checks if it's set, if so it will set `more_params` to `true` and also assign the string `odom_topic` the name of the topic for `odom_x`.

There might be a nicer way to get rid of one `get_parameter` call, not sure how.

Ok a tiny bit convoluted explanation, but in short it's an attempt to reproduce the assigning to bool if a parameter is set or not.

Reply:
Well the only bad thing is you're now declaring the `_N+1`th parameter when it doesnt exist. I'd say you could make an `else` to `undeclare` it, but that's not good posture either. 

There are documented parameters for `  node_options.allow_undeclared_parameters(true);` in the node options object to grab parameters from a file before declaring them, maybe this is a good use-case for it. I'd leave it up to @ayrton04 if he thinks that declaring, checking, and then undeclaring is an acceptable workaround. It seems a little hacky for my blood but I'd accept it to get this PR merged with a ticket to fix later

Reply:
I tend to agree that this feels way to hacky, although I also did not want to use the `node_options.allow_undeclared_parameters(true);` as it seems like a good candidate for future deprecation. 
I'm fine either way, we'll see what Tom says, perhaps a later ticket for fix is good enough for now.

Reply:
Actually from a discussion in another thread with Tully, Ithink that undeclared parameters thing is going to stay for good. They claim there's a few good reasons for it. 

If we keep it this way, if its unset, you should `undeclare_parameter` that parameters so that when the user goes `ros2 topic list` it doesnt appear

Reply:
I think I might have come across that thread, at least a very similar discussion.

I added the else condition for undeclaring the parameters.